sandman/source/schedule.cpp

```diff
@@ -50,80 +41,48 @@ struct ScheduleEvent
 // Whether the system is initialized.
 static bool s_ScheduleInitialized = false;
 
-// This will contain the schedule once it has been parsed in.
-static std::vector<ScheduleEvent> s_ScheduleEvents;
-
 // The current spot in the schedule.
 static unsigned int s_ScheduleIndex = UINT_MAX;
 
 // The time the delay for the next event began.
 static Time s_ScheduleDelayStartTime;
 
+static Schedule s_Schedule;
+
 // Functions
 //
 
+// Schedule members
 
-// ScheduleEvent members
-
-// Read a schedule event from JSON. 
-//	
-// p_Object:	The JSON object representing the event.
-//
-// Returns:		True if the event was read successfully, false otherwise.
-//
-bool ScheduleEvent::ReadFromJSON(rapidjson::Value::ConstObject const& p_Object)
+void Schedule::AddEvent(const ScheduleEvent& p_event)
 {
-	// We must have a delay.
-	auto const l_DelayIterator = p_Object.FindMember("delaySec");
-
-	if (l_DelayIterator == p_Object.MemberEnd())
-	{
-		return false;
-	}
-
-	if (l_DelayIterator->value.IsInt() == false)
-	{
-		return false;
-	}
-
-	m_DelaySec = l_DelayIterator->value.GetInt();
+	m_ScheduleEvents.push_back(p_event);
+}
 
-	// We must also have a control action.
-	auto const l_ControlActionIterator = p_Object.FindMember("controlAction");
+bool Schedule::IsEmpty()
+{
+	return (m_ScheduleEvents.size() == 0);
+}
 
-	if (l_ControlActionIterator == p_Object.MemberEnd())
-	{
-		return false;
-	}
+size_t Schedule::GetNumEvents() const
+{
+	return m_ScheduleEvents.size();
+}
 
-	if (l_ControlActionIterator->value.IsObject() == false)
-	{
-		return false;
-	}
-	
-	if (m_ControlAction.ReadFromJSON(l_ControlActionIterator->value.GetObject()) == false) 
-	{
-		return false;
-	}
-	
-	return true;
+std::vector<ScheduleEvent> Schedule::GetEvents() const
```

Comment:
See declaration.

Reply:
Okay, this isn't quite right. It still makes a copy of the list. Since we need to be able to modify the controls in the control actions, this can't be const, but it can still be a reference. So this line should look like

std::vector<ScheduleEvent>& Schedule::GetEvents()

for now.